### PR TITLE
Port emulator to aglib api

### PR DESCRIPTION
**Pull request by patryk4815**
done

### CODE REVIEW

File: pwndbg/aglib/disasm/__init__.py [hunk 1]
```diff
@@ -308,8 +302,16 @@ def can_run_first_emulate() -> bool:
     try:
         from mmap import mmap, MAP_ANON, MAP_PRIVATE, PROT_EXEC, PROT_READ, PROT_WRITE  # isort:skip
 
+        access_mode = PROT_WRITE | PROT_READ | PROT_EXEC
+        if sys.platform == "darwin":
+            # On macOS (Darwin), creating mmap with 'rwx' permissions is blocked
+            # by System Integrity Protection (SIP), which prevents execution of code
+            # from writable and readable memory regions. Therefore, we restrict
+            # the access mode to 'rw' (read and write only).
+            access_mode = PROT_WRITE | PROT_READ
```
Comment (disconnect3d) [inline pwndbg/aglib/disasm/__init__.py:311]: How does UnicornEngine works on macOS/darwin then? Does it not allocate an RWX memory page on macOS? How do they JIT the code then?

File: pwndbg/emu/emulator.py [hunk 2]
```diff
@@ -574,16 +577,19 @@ def get_uc_mode(self):
...
 
         elif arch in ("arm", "aarch64"):
-            mode |= U.UC_MODE_THUMB if (pwndbg.gdblib.regs.cpsr & (1 << 5)) else U.UC_MODE_ARM
+            mode |= U.UC_MODE_THUMB if (pwndbg.aglib.regs.cpsr & (1 << 5)) else U.UC_MODE_ARM
 
-        elif arch == "mips" and "isa32r6" in gdb.newest_frame().architecture().name():
+        elif (
+            arch == "mips"
+            and pwndbg.dbg.is_gdblib_available()
+            and "isa32r6" in gdb.newest_frame().architecture().name()
```
Comment (disconnect3d) [inline pwndbg/emu/emulator.py:590]: Why gdb check here? does lldb not support mips?

File: pwndbg/aglib/disasm/__init__.py (hunk 1 above)
Comment (patryk4815) [reply pwndbg/aglib/disasm/__init__.py:311]: I think Unicorn works fine. If you need executable page, first you create `rw`, then change it with mprotect to `rx`.
Only `wx` is not permitted.

image

File: pwndbg/emu/emulator.py (hunk 2 above)
Comment (patryk4815) [reply pwndbg/emu/emulator.py:590]: We'll address the `gdb.newest_frame().architecture().name()` calls later as they appear in many places. For now, this is a 'best effort' implementation 🤔

Comment (patryk4815) [reply pwndbg/emu/emulator.py:590]: We just here don't support `MIPS32R6` arch for lldb

**Pull request merged**